app/views/course/announcements/_form.html.slim

```diff
@@ -0,0 +1,12 @@
+= simple_form_for [@course, @announcement],
+                    url: (@announcement.new_record? ?  course_announcements_url(@course) : \
+                         course_announcement_url(@course, @announcement)) do |f|
```

Comment:
Form_for cannot generate urls helpers correctly, it expects <code>course_course_announcement_path</code> while we don't have.

I think this is issue with scoped routes, to fix I found 2 solutions: 
1. manually specify the url
2. override <code>self.model_name</code> in announcement model

REF: 
http://stackoverflow.com/questions/11332657/rails-routing-scope-namespace-problems
https://www.ruby-forum.com/topic/2385699